### DISCUSSION TITLE
Welcome to lightbug_http Discussions!

### DISCUSSION THREAD

**Discussion by saviorand** [saviorand/lightbug_http#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ortega225** [comment]
Howdy,
    Coming in from the PHP and Node JS world for backend and Python for Data Analytics / Machine Learning, looking forward to Mojo!

A few questions for roadmap and future features:

1. Will Lightbug API support asynchronous code?
- I didn't see async / await syntax in your examples, so I wasn't sure 
- With an API, asynchronous is vital to prevent things like a DB call or an external API call holding up other calls to the API

2. Will Lightbug API have an built in ORM?
- Coming from PHP Laravel, their ORM is magical and I've had a similar experience with Adonis JS, but the ORM options for Python seem to be not as polished. ( https://orm.masoniteproject.com/ looks interesting, but I haven't tried it in production)
- Having support for migrations, models, and seeders would be fantastic as well

3. Will Lightbug API be MVC framework?

**Comment by saviorand** [comment]
@ortega225 howdy! Great questions, here are my thoughts so far, it's all very much in the design stage though so curious what your and others thoughts are on these points as well:
1. Yes definitely, async is a must-have! The reason there is no async in Lightbug yet in general is because async capabilities are still a work in progress in the Mojo language itself. It wouldn't make sense to build on the current implementation as there's a chance it will change soon. There is no built-in async runtime in Mojo for example. I was thinking about making something like a basic async runtime myself but there's basically no documentation on async capabiltiies in general and not much in terms of async examples in Mojo which was a blocker for me personally so far to introduce this. 
2. I do want to add a way to work with DBs from Lightbug, most likely starting with Postgres. Although so far there's not much in terms of libraries to build this on, even a Postgres driver implementation is not something that exists in Mojo (yet). In my opinion an ORM would have to be a standalone project which Lightbug would import and maybe have some deeper integration with. Without a DB driver in Mojo one idea I had was to use something like [PostgREST](https://docs.postgrest.org/en/v12/) to implement an ORM with HTTP calls, but didn't start actively working on this yet
3. I don't think Lightbug will explicitly implement MVC, but in some form the parts of it would be present, e.g we'll definitely need a way to separate entities/domain logic from app logic. What's nice with Mojo is that the Model part could be made very readable with the Python-like syntax, making it's easier to reason about the underlying logic in the app.
